jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/DefaultQueryRewriter.java

```diff
@@ -139,13 +143,24 @@ public String rewriteFilterItem(FilterItem item) {
                     }
                 }
 
-                FilterItem replacementFilterItem = new FilterItem(item.getSelectItem(), item.getOperator(), elements);
+                FilterItem replacementFilterItem = new FilterItem(selectItem, item.getOperator(), elements);
                 return super.rewriteFilterItem(replacementFilterItem);
             }
         }
         return super.rewriteFilterItem(item);
     }
-    
+
+    protected String rewriteTimestamp(FilterItem item) {
```

Comment:
Now this method does a rewrite of a whole filter item and not just the timestamp literal. I was suggesting to make something that would make the burden of overriding as little as possible, so that you would only need to do like this:

```
protected String rewriteTimestamp(Timestamp timestamp) {
  return MyStrangeTimestampFormat.format(timestamp);
} 
```